### PR TITLE
Use TF-native NMS when in TF backend

### PR DESCRIPTION
**Pull request by ianstenbit**

### CODE REVIEW

**Review by tirthasheshpatel — APPROVED** [review]
Looks good! Thanks for jumping in and fixing this!

File: keras_cv/layers/object_detection/non_max_suppression.py [hunk 1]
```diff
@@ -88,8 +88,17 @@ def call(
...
+                max_output_size=self.max_detections,
+                iou_threshold=self.iou_threshold,
+                score_threshold=self.confidence_threshold,
+                pad_to_max_output_size=True,
+                sorted_input=False,
+            )
+        elif keras.backend.backend() == "torch":
             # Since TorchVision has a nice efficient NMS op, we might as well
             # use it!
             if keras.backend.backend() == "torch":
```
Comment (jbischof) [inline keras_cv/layers/object_detection/non_max_suppression.py:104]: We are checking this condition twice now

**Review by jbischof — APPROVED** [review]
Thanks!

File: keras_cv/layers/object_detection/non_max_suppression.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/layers/object_detection/non_max_suppression.py]: Welp that's embarrassing -- fixed

**Pull request merged**